### PR TITLE
Expose `options` parameter to @turf/boolean-intersects and @turf/booean-disjoint to add option of ignoring self intersections.

### PR DESCRIPTION
**Pull request by 01100100**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Have read [How To Contribute](https://github.com/Turfjs/turf/blob/master/CONTRIBUTING.md#how-to-contribute).
- [x] Run npm test at the sub modules where changes have occurred.
- [x] Run npm run lint to ensure code style at the turf module level.

As mentioned in [https://github.com/Turfjs/turf/issues/2633](https://github.com/Turfjs/turf/issues/2633)

The @turf/boolean-intersects function does not handle cases when the user wants to ignore self intersections.

This PR introduces the changes needed to accommodate the `options.ignoreSelfIntersections` parameter.
 
**MODIFIED PACKAGES:**

* boolean-disjoint
* boolean-intersects

The changes are backwards compatible.

I would like to include some extra tests for the new cases, but not sure how to best add/structure tests for this optional parameter, given the current setup [https://github.com/Turfjs/turf/blob/master/packages/turf-boolean-intersects/test.ts#L9](https://github.com/Turfjs/turf/blob/master/packages/turf-boolean-intersects/test.ts#L9)

Below is an example of a lineString containing a self intersection, that might be useful for testing the changes introduced in the PR.

```json
{
  "type": "Feature",
  "properties": {},
  "geometry": {
    "type": "LineString",
    "coordinates": [
      [
        1,
        1
      ],
      [
        2,
        2
      ],
      [
        1,
        2
      ],
      [
        2,
        1
      ]
    ]
  }
}
```

### CODE REVIEW

**Comment by smallsaucepan** [comment]
Thanks heaps for putting this together @01100100. Before we look to merge, could you please add a couple of tests to cover passing true, false or empty for this new option?

**Comment by 01100100** [comment]
@smallsaucepan added the test cases and checked they all passed as expected :+1: 

Let me know if there is anything else that this PR needs!

**Comment by 01100100** [comment]
Hi @smallsaucepan and other maintainers :wave: 

Anything I can do to help this getting merged and in a release? :safety_vest:

**Comment by 01100100** [comment]
Hi @smallsaucepan and other maintainers :wave: 

Anything I can do to help this getting merged and in a release? I have a monkey patched :monkey: :wrench:  version of this function working in a project, and would like to use turf.js without modifying it. Happy to help if something else needs to be done here :smile:

**Comment by smallsaucepan** [comment]
Thanks for following up @01100100. Also added some review comments on boolean-disjoint/index.ts about tightening up the types. Do those show up for you?

**Comment by smallsaucepan** [comment]
My apologies. I had added comments but they were only "pending". Please see if they're visible now.

**Comment by 01100100** [comment]
All good :smile: I can see them now, will have a look later and ping you when I am done. Thanks and have a good one!

File: packages/turf-boolean-disjoint/index.ts
Comment (01100100) [reply packages/turf-boolean-disjoint/index.ts]: Done, cheers for the ref material :+1:

File: packages/turf-boolean-disjoint/index.ts
Comment (01100100) [reply packages/turf-boolean-disjoint/index.ts:33]: I am not sure the best way to do this. I would appreciate some advise if you know a nicer way.

For now I put a extra  `documentedOptions` const in, but see how that only caters for single for a single option key (and in the future more might be provided).

```ts
  const documentedOptions = options.ignoreSelfIntersections
    ? { ignoreSelfIntersections: options.ignoreSelfIntersections }
    : {};
 ```
 
 Could you please show me a better way of doing this?

**Comment by 01100100** [comment]
Hi @smallsaucepan + others,

Just chasing this up as its still open. Is there anything else that needs to be done? Would be more then happy to help

File: packages/turf-boolean-disjoint/index.ts
Comment (smallsaucepan) [reply packages/turf-boolean-disjoint/index.ts:33]: Missed this comment sorry. I've got another way to propose, which I'll push to your branch if you don't mind? Think it reads a bit better.

**Comment by smallsaucepan** [comment]
Hi @01100100. I've got an alternative way to do the options passing. You mind if I push a commit to your branch to demonstrate?

**Comment by 01100100** [comment]
Hi @smallsaucepan

Yes please do! I would appreciate seeing a nice way to do this

**Comment by smallsaucepan** [comment]
@01100100 changed tack slightly with the options, so instead we pass ignoreSelfIntersections around internally as a boolean, and only worry about checking / packaging is as an options object when viewed from the outside. Let me know if you think that works and we can press on.

**Review by smallsaucepan — APPROVED** [review]

**Comment by 01100100** [comment]
Hi @smallsaucepan, 

That is a much more concise way of doing it. Thanks for sharing. Works for me and happy if its merged :smile:

**Comment by coracuity** [comment]
I ran into this yesterday upgrading to 7.0, I'm very excited for this to be merged/released!

**Comment by 01100100** [comment]
Hi @smallsaucepan,

Let me know if you need anything more done here and thanks for maintaining this project! 

Regards,
Dave

**Comment by smallsaucepan** [comment]
Thanks for working through this with us @01100100.

**Pull request merged**
